app/code/community/Clean/SqlReports/Model/Report.php

```diff
@@ -64,4 +64,35 @@ public function getGridConfig()
         }
         return $this->_gridConfig;
     }
-}
\ No newline at end of file
+
+    /**
+     * Disallow TRUNCATE, DROP, DELETE statements & remove semicolon terminator
+     */
+    protected function _beforeSave() {
+        $disallowedPatterns = array(
+            'TRUNCATE TABLE',
+            'DROP TABLE',
+            'DROP TEMPORARY TABLE',
+            'DELETE FROM'
+        );
+		
+        $sqlQuery = $this->getSqlQuery();
+
+        if (substr($sqlQuery, -1) === ';') {
+            $this->setSqlQuery(substr($sqlQuery, 0, -1));
+            Mage::getSingleton('adminhtml/session')->addNotice(Mage::helper('cleansql')->__('Do not include a semicolon terminator'));
+        }
+
+        foreach ($disallowedPatterns as $pattern) {
```

Comment:
I don't think it's worth the effort to try to prevent destructive queries. There's bound to be many ways around it and I doubt we want to manage some blacklist of patterns. For instance, your approach would not catch `TRUNCATE    TABLE` (multiple spaces). You could solve some white space issues via regex but `TRUNCATE /*some comment*/ TABLE` would still get through. 

The best approach is to create a use with read-only access to the database. We could highlight this better in the `README.md` and perhaps provide some guidance.